web/client/components/map/openlayers/MeasurementSupport.jsx

```diff
@@ -65,13 +104,56 @@ class MeasurementSupport extends React.Component {
         return null;
     }
 
+    isPolygon = (feature) => {
+        return feature.geometry.type === "Polygon";
+    }
+    updateFeatures = (props) => {
+        let ft = props.measurement.feature;
+        if (props.measurement.lineMeasureEnabled) {
+            let newCoords = transformLineToArcs(props.measurement.feature.geometry.coordinates);
+            ft = set("geometry.coordinates", newCoords, ft);
+        }
+        this.replaceFeatures([ft]);
+
+        if (props.measurement.showLabel) {
+            this.removeMeasureTooltips();
+            this.measureTooltipElement = document.createElement("div");
+            this.measureTooltipElement.className = this.drawing ? "tooltip tooltip-measure" : "tooltip tooltip-static";
+
+            let geom = this.source.getFeatures()[0].getGeometry();
+            let output;
+            if (geom instanceof ol.geom.Polygon) {
+                output = this.formatArea(geom, props);
+                this.tooltipCoord = geom.getInteriorPoint().getCoordinates();
+            } else if (geom instanceof ol.geom.LineString) {
+                output = this.formatLength(geom, props);
+                this.tooltipCoord = geom.getLastCoordinate();
+            }
+            this.measureTooltipElement.innerHTML = output;
+            this.measureTooltip = new ol.Overlay({
+                element: this.measureTooltipElement,
+                offset: [0, -7],
+                positioning: 'bottom-center'
+            });
+            this.measureTooltip.setPosition(this.tooltipCoord);
+            props.map.addOverlay(this.measureTooltip);
+        }
+        this.updateMeasurementResults(props, this.source.getFeatures()[0]);
```

Comment:
but I need to know that the update has been made via UI (changing of feature geometry) because then i'll update the state props and then this flag (updatebyUI) it is checked via willReceiveProps in order to avoid endless loop

Reply:
Correct me if I'm wrong.

I see the second argument of `updateMeasurementResults` is used **only** here and that argument, called  `olFeatureUpdateByUI`, is used, inside the function definition, only to do this:
```javascript
let olFeatureToUse = this.sketchFeature;
        if (!this.sketchFeature) {
            return;
        }
        if (olFeatureUpdateByUI) {
            olFeatureToUse = olFeatureUpdateByUI;
            this.sketchFeature = olFeatureUpdateByUI;
        }
```
This means that `olFeatureByUI`, that is `this.source.getFeatures()[0]` is assigned to `this.sketchFeature`, if present, and used for nothing else. olFeatureToUse is always this.sketchFeature, so the 2 variables can be removed. 

The only corner case is when this.source.getFeatures()[0] != null and this.sketchFeature = null. You can solve this corner case outside the function, without complicating the logic or adding non-sense arguments to the function.